### ISSUE TITLE
Arming/disarming with PIN

### ISSUE THREAD

**Issue by russilker** [Blackymas/NSPanel_HA_Blueprint#1774]
### TFT Version

4.2.4

### ESPHome Version

4.2.4

### Blueprint Version

4.2.4

### Panel Model

NSPanel US Model

### What is the bug?

Alarm won't arm/disarm with PIN

### Steps to Reproduce

Try to arm/disarm alarm using PIN. Entering the PIN does nothing, numpad stays on screen, no logs in Home Assistant, no errors in ESPHome logs after trying to submit PIN.

### Your panel's YAML

```yaml
substitutions:

###### CHANGE ME START ######

  device_name: "switch_entryway_nspanel" 
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password

  nextion_update_url: "https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_us.tft" # URL to Github

##### CHANGE ME END #####



##### DO NOT CHANGE ANYTHING! #####

packages:
  ##### download esphome code from Github
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files: [nspanel_esphome.yaml]
    refresh: 300s

##### DO NOT CHANGE ANYTHING! #####
```

### ESPHome logs

```logs
[21:51:13][D][text_sensor:064]: 'switch_entryway_nspanel NSPanel local event': Sending state '{"page": "alarm", "key": "disarm", "value": "click", "entity": "", "mui": "Disarm", "code_format": "number", "code_arm_req": "0"}'
[21:51:13][D][nextion_textsensor:014]: Processed text_sensor "localevent" state "{"page": "alarm", "key": "disarm", "value": "click", "entity": "", "mui": "Disarm", "code_format": "number", "code_arm_req": "0"}"
[21:51:14][D][nextion:464]: Got new page: 24
[21:51:14][D][display.disp1.on_page:716]: Nextion page changed
[21:51:14][D][display.disp1.on_page:717]: New page: keyb_num (24)
[21:51:14][D][text_sensor:064]: 'switch_entryway_nspanel Current page': Sending state 'keyb_num'
[21:51:14][D][script.page_changed:2110]: New page: keyb_num
[21:51:14][D][script.page_changed:2111]: Entity shown: alarm_control_panel.********
[21:51:14][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:14][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:14][W][nextion:405]: Nextion reported variable name invalid!
[21:51:14][D][sensor:094]: 'switch_entryway_nspanel Page Id': Sending state 24.00000  with 0 decimals of accuracy
[21:51:14][D][sensor.page_id:1381]: New page Id: 24
[21:51:14][W][component:214]: Component display took a long time for an operation (0.05 s).
[21:51:14][W][component:215]: Components should block for at most 20-30ms.
[21:51:15][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:15][D][ledc.output:060]: Resolution calculated as 16
[21:51:15][D][ledc.output:055]: Calculating resolution bit-depth for frequency 1047.000000
[21:51:15][D][ledc.output:060]: Resolution calculated as 16
[21:51:15][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:15][D][ledc.output:060]: Resolution calculated as 16
[21:51:15][D][nextion:446]: Got touch event:
[21:51:15][D][nextion:447]:   page_id:      24
[21:51:15][D][nextion:448]:   component_id: *
[21:51:15][D][nextion:449]:   event type:   PRESS
[21:51:15][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:15][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:15][D][ledc.output:055]: Calculating resolution bit-depth for frequency 880.000000
[21:51:15][D][ledc.output:060]: Resolution calculated as 16
[21:51:15][D][nextion:446]: Got touch event:
[21:51:15][D][nextion:447]:   page_id:      24
[21:51:15][D][nextion:448]:   component_id: *
[21:51:15][D][nextion:449]:   event type:   RELEASE
[21:51:15][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:15][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:15][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:15][D][ledc.output:060]: Resolution calculated as 16
[21:51:15][D][nextion:446]: Got touch event:
[21:51:15][D][nextion:447]:   page_id:      24
[21:51:15][D][nextion:448]:   component_id: *
[21:51:15][D][nextion:449]:   event type:   PRESS
[21:51:15][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:15][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:15][D][nextion:446]: Got touch event:
[21:51:15][D][nextion:447]:   page_id:      24
[21:51:15][D][nextion:448]:   component_id: *
[21:51:15][D][nextion:449]:   event type:   RELEASE
[21:51:15][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:15][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:16][D][nextion:446]: Got touch event:
[21:51:16][D][nextion:447]:   page_id:      24
[21:51:16][D][nextion:448]:   component_id: *
[21:51:16][D][nextion:449]:   event type:   PRESS
[21:51:16][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:16][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:16][D][nextion:446]: Got touch event:
[21:51:16][D][nextion:447]:   page_id:      24
[21:51:16][D][nextion:448]:   component_id: *
[21:51:16][D][nextion:449]:   event type:   RELEASE
[21:51:16][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:16][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:16][D][nextion:446]: Got touch event:
[21:51:16][D][nextion:447]:   page_id:      24
[21:51:16][D][nextion:448]:   component_id: *
[21:51:16][D][nextion:449]:   event type:   PRESS
[21:51:16][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:16][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:16][D][nextion:446]: Got touch event:
[21:51:16][D][nextion:447]:   page_id:      24
[21:51:16][D][nextion:448]:   component_id: *
[21:51:16][D][nextion:449]:   event type:   RELEASE
[21:51:16][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:16][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:17][D][nextion:446]: Got touch event:
[21:51:17][D][nextion:447]:   page_id:      24
[21:51:17][D][nextion:448]:   component_id: 15
[21:51:17][D][nextion:449]:   event type:   PRESS
[21:51:17][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:17][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:17][D][rtttl:176]: Playback finished
[21:51:17][D][nextion:446]: Got touch event:
[21:51:17][D][nextion:447]:   page_id:      24
[21:51:17][D][nextion:448]:   component_id: 15
[21:51:17][D][nextion:449]:   event type:   RELEASE
[21:51:17][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:17][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:17][D][rtttl:051]: Playing song FinalCountdown
[21:51:17][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:17][D][ledc.output:060]: Resolution calculated as 16
[21:51:18][D][ledc.output:055]: Calculating resolution bit-depth for frequency 880.000000
[21:51:18][D][ledc.output:060]: Resolution calculated as 16
[21:51:18][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:18][D][ledc.output:060]: Resolution calculated as 16
[21:51:18][D][ledc.output:055]: Calculating resolution bit-depth for frequency 659.000000
[21:51:18][D][ledc.output:060]: Resolution calculated as 16
[21:51:19][D][nextion:446]: Got touch event:
[21:51:19][D][nextion:447]:   page_id:      24
[21:51:19][D][nextion:448]:   component_id: 15
[21:51:19][D][nextion:449]:   event type:   PRESS
[21:51:19][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:19][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:19][D][ledc.output:055]: Calculating resolution bit-depth for frequency 1047.000000
[21:51:19][D][ledc.output:060]: Resolution calculated as 16
[21:51:19][D][nextion:446]: Got touch event:
[21:51:19][D][nextion:447]:   page_id:      24
[21:51:19][D][nextion:448]:   component_id: 15
[21:51:19][D][nextion:449]:   event type:   RELEASE
[21:51:19][D][script:077]: Script 'timer_page' restarting (mode: restart)
[21:51:19][D][script:077]: Script 'timer_dim' restarting (mode: restart)
[21:51:20][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:20][D][ledc.output:060]: Resolution calculated as 16
[21:51:20][D][ledc.output:055]: Calculating resolution bit-depth for frequency 1047.000000
[21:51:20][D][ledc.output:060]: Resolution calculated as 16
[21:51:20][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:20][D][ledc.output:060]: Resolution calculated as 16
[21:51:20][D][ledc.output:055]: Calculating resolution bit-depth for frequency 880.000000
[21:51:20][D][ledc.output:060]: Resolution calculated as 16
[21:51:21][D][ledc.output:055]: Calculating resolution bit-depth for frequency 1047.000000
[21:51:21][D][ledc.output:060]: Resolution calculated as 16
[21:51:22][D][ledc.output:055]: Calculating resolution bit-depth for frequency 988.000000
[21:51:22][D][ledc.output:060]: Resolution calculated as 16
[21:51:22][D][ledc.output:055]: Calculating resolution bit-depth for frequency 1047.000000
[21:51:22][D][ledc.output:060]: Resolution calculated as 16
[21:51:22][D][ledc.output:055]: Calculating resolution bit-depth for frequency 659.000000
[21:51:22][D][ledc.output:060]: Resolution calculated as 16
[21:51:23][D][ledc.output:055]: Calculating resolution bit-depth for frequency 880.000000
[21:51:23][D][ledc.output:060]: Resolution calculated as 16
[21:51:24][D][ledc.output:055]: Calculating resolution bit-depth for frequency 784.000000
[21:51:24][D][ledc.output:060]: Resolution calculated as 16
[21:51:24][D][ledc.output:055]: Calculating resolution bit-depth for frequency 880.000000
[21:51:24][D][ledc.output:060]: Resolution calculated as 16
```

### Home Assistant logs

```logs
Nothing during this time
```

**Comment by Doughboy68** [comment]
I reported this already and is supposed to be fixed in dev. I have not had a chance to try yet. I confirmed. No issue with landscape us panel so I believe was something missed.

**Comment by edwardtfn** [comment]
- #1705 
- #1766

**Comment by russilker** [comment]
Thanks, sorry didn't see it when I searched.

**Comment by edwardtfn** [comment]
We just released a patch with solves this: [v4.2.5 - Celebrating 1000 Stars with Enhanced Functionality and UI Improvements](https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v4.2.5)
Could you please update your system and let me know if the issue was solved?
